qiskit/chemistry/ground_state_calculation/mes_ground_state_calculation.py

```diff
@@ -53,27 +55,45 @@ def returns_groundstate(self) -> bool:
 
         return False
 
-    def compute_groundstate(self, driver: BaseDriver) -> EigenstateResult:
+    def compute_groundstate(self, driver: BaseDriver,
+                            aux_operators: Optional[List[Any]] = None
+                            ) -> EigenstateResult:
         """Compute Ground State properties.
 
         Args:
             driver: a chemistry driver object which defines the chemical problem that is to be
                     solved by this calculation.
+            aux_operators: Additional auxiliary operators to evaluate at the ground state.
+                Depending on whether a fermionic or bosonic system is solved, the type of the
+                operators must be ``FermionicOperator`` or ``BosonicOperator``, respectively.
+
+        Raises:
+            NotImplementedError: If an operator in ``aux_operators`` is not of type
+                ``FermionicOperator``.
 
         Returns:
             An eigenstate result. Depending on the transformation this can be an electronic
             structure or bosonic result.
         """
-        operator, aux_operators = self.transformation.transform(driver)
+        if aux_operators is not None:
+            if any(not isinstance(op, (WeightedPauliOperator, FermionicOperator))
+                   for op in aux_operators):
+                raise NotImplementedError('Currently only fermionic problems are supported.')
+
+        # get the operator and auxiliary operators, and transform the provided auxiliary operators
+        # note that ``aux_operators`` contains not only the transformed ``aux_operators`` passed
+        # by the user but also additional ones from the transformation
+        operator, aux_operators = self.transformation.transform(driver, aux_operators)
 
         if isinstance(self._solver, MESFactory):
             # this must be called after transformation.transform
             solver = self._solver.get_solver(self.transformation)
         else:
             solver = self._solver
 
-        aux_operators = aux_operators if solver.supports_aux_operators() else None
-
+        # convert aux_operators to a list for the minimum eigensolver
+        if not solver.supports_aux_operators():
+            aux_operators = None
```

Comment:
Shouldn't we just keep this like it was?

Reply:
It's a new file anyways so this will show up in the changelog either way, right? 😄 This way is a bit more explicit which is nice to read, or do you have a strong opinion for the one-liner?

Reply:
Ah no it's fine then. It just stood out as an unnecessary change but yes, since it's a new file in Qiskit once merged, it really doesn't matter :+1: